TestPackedInts.testPackedLongValues failing

### Description

TestPackedInts.testPackedLongValues fails repeatably, example trace:

```
org.apache.lucene.util.packed.TestPackedInts > testPackedLongValues FAILED
    java.lang.AssertionError: expected:<2368> but was:<2384>
        at __randomizedtesting.SeedInfo.seed([E73C4D01E8CF65EC:E32957873BE4F660]:0)
        at org.junit.Assert.fail(Assert.java:89)
        at org.junit.Assert.failNotEquals(Assert.java:835)
        at org.junit.Assert.assertEquals(Assert.java:647)
        at org.junit.Assert.assertEquals(Assert.java:633)
        at org.apache.lucene.util.packed.TestPackedInts.testPackedLongValues(TestPackedInts.java:1022)
```

### Gradle command to reproduce

./gradlew :lucene:core:test --tests "org.apache.lucene.util.packed.TestPackedInts.testPackedLongValues" -Ptests.jvms=2 -Ptests.jvmargs= -Ptests.seed=E73C4D01E8CF65EC -Ptests.gui=true -Ptests.file.encoding=US-ASCII -Ptests.vectorsize=128

--- a/lucene/core/src/java/org/apache/lucene/util/packed/PackedInts.java
+++ b/lucene/core/src/java/org/apache/lucene/util/packed/PackedInts.java
@@ -604,8 +604,10 @@ public class PackedInts {
 
     @Override
     public long ramBytesUsed() {
-      return RamUsageEstimator.alignObjectSize(
-          RamUsageEstimator.NUM_BYTES_OBJECT_HEADER + Integer.BYTES);
+      return valueCount == PackedLongValues.DEFAULT_PAGE_SIZE
+          ? 0
+          : RamUsageEstimator.alignObjectSize(
+              RamUsageEstimator.NUM_BYTES_OBJECT_HEADER + Integer.BYTES);
     }
   }
 

--- a/lucene/core/src/test/org/apache/lucene/util/packed/TestPackedInts.java
+++ b/lucene/core/src/test/org/apache/lucene/util/packed/TestPackedInts.java
@@ -18,12 +18,15 @@ package org.apache.lucene.util.packed;
 
 import com.carrotsearch.randomizedtesting.generators.RandomNumbers;
 import java.io.IOException;
+import java.lang.reflect.Field;
 import java.nio.ByteBuffer;
 import java.nio.LongBuffer;
 import java.util.ArrayList;
 import java.util.Arrays;
+import java.util.Collection;
 import java.util.List;
 import java.util.Locale;
+import java.util.Map;
 import java.util.Random;
 import org.apache.lucene.store.ByteArrayDataInput;
 import org.apache.lucene.store.DataInput;
@@ -966,6 +969,18 @@ public class TestPackedInts extends LuceneTestCase {
             .ramBytesUsed());
   }
 
+  private static final class IgnoreNullReaderSingletonAccumulator
+      extends RamUsageTester.Accumulator {
+    @Override
+    public long accumulateObject(
+        Object o, long shallowSize, Map<Field, Object> fieldValues, Collection<Object> queue) {
+      if (o == PackedInts.NullReader.forCount(PackedLongValues.DEFAULT_PAGE_SIZE)) {
+        return 0;
+      }
+      return super.accumulateObject(o, shallowSize, fieldValues, queue);
+    }
+  }
+
   public void testPackedLongValues() {
     final long[] arr =
         new long[RandomNumbers.randomIntBetween(random(), 1, TEST_NIGHTLY ? 1000000 : 10000)];
@@ -1017,7 +1032,8 @@ public class TestPackedInts extends LuceneTestCase {
         for (int i = 0; i < arr.length; ++i) {
           buf.add(arr[i]);
           if (rarely() && !TEST_NIGHTLY) {
-            final long expectedBytesUsed = RamUsageTester.ramUsed(buf);
+            final long expectedBytesUsed =
+                RamUsageTester.ramUsed(buf, new IgnoreNullReaderSingletonAccumulator());
             final long computedBytesUsed = buf.ramBytesUsed();
             assertEquals(expectedBytesUsed, computedBytesUsed);
           }
@@ -1044,7 +1060,8 @@ public class TestPackedInts extends LuceneTestCase {
         }
         assertFalse(it.hasNext());
 
-        final long expectedBytesUsed = RamUsageTester.ramUsed(values);
+        final long expectedBytesUsed =
+            RamUsageTester.ramUsed(values, new IgnoreNullReaderSingletonAccumulator());
         final long computedBytesUsed = values.ramBytesUsed();
         assertEquals(expectedBytesUsed, computedBytesUsed);
       }

